backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java

```diff
@@ -122,15 +119,22 @@ public void logArtifacts(Handle handle, T entityId, List<Artifact> artifacts, bo
       validateAndThrowErrorAlreadyExistsArtifacts(entityId, artifacts, handle);
     }
 
-    for (final var artifact : artifacts) {
+    for (var artifact : artifacts) {
+      var path = artifact.getPath();
       var uploadCompleted = !artifactStoreConfig.getArtifactStoreType().equals(CommonConstants.S3);
       if (artifact.getUploadCompleted()) {
         uploadCompleted = true;
+      } else {
+        path =
+            artifactStoreConfig.storeTypePathPrefix()
+                + this.entityName
+                + "/"
+                + entityId
+                + "/"
+                + artifact.getKey();
```

Comment:
Let's make sure to do some input hygiene since this value is provided directly by the user. That's _always_ a dangerous thing to just pass along. Maybe base64 it so it's always safe?

Reply:
@conradoverta For regular `get_`/`log_artifact()`, no the S3 final location name doesn't matter—the client just cares about the artifact key and the pre-signed URL.

But as I took a deeper dive into the client codebase, and there are some considerations I hadn't realized before 😰 :

1. In `fetch_artifacts()`, client [uses](https://github.com/VertaAI/modeldb/blob/299d4525bb579cf58b257cbbd47f475750770d77/client/verta/verta/tracking/entities/_deployable_entity.py#L438-L440) the artifact proto's `path` field as a filepath to locally cache the downloaded artifact (b/c it's a checksum). This shouldn't be an issue though; client doesn't care what it is as long as it's unique.
2. The ZFS/local artifacts functionality we built a couple years ago involves [prepending `$VERTA_ARTIFACT_DIR` to the artifact proto's `path`](https://github.com/VertaAI/modeldb/blob/36a9000607a7e9635e8cf7d3684f3a7a57e9cc7d/client/verta/verta/tracking/entities/_experimentrun.py#L235-L245). Also, there's a `run.log_artifact_path()` that explicitly relies on [using the artifact proto's `path`.](https://github.com/VertaAI/modeldb/blob/36a9000607a7e9635e8cf7d3684f3a7a57e9cc7d/client/verta/verta/tracking/entities/_experimentrun.py#L393). I wish I knew whether these are still being actively used 🤔

Reply:
1. yep, it should still be unique
2. I'm not particularly concerned about these

Reply:
👌